sdk/azidentity/live_test.go

```diff
@@ -272,7 +284,16 @@ func (p *recordingPolicy) Do(req *policy.Request) (resp *http.Response, err erro
 		r.URL.Host = r.Host
 		r.URL.Scheme = "https"
 	}
-	return req.Next()
+	res, err := req.Next()
+	if err == nil {
+		// if the response is a recording mismatch, return it as a simple error that prints clearly in the test log
+		if e := res.Header.Get("x-request-mismatch-error"); e != "" {
```

Comment:
This seems duplicated from your same change in `/sdk/internal`. I guess that hasn't been released yet though. Will this go away once that happens?

Reply:
Yes, I'll remove this before merging